Plandomizer.py

```diff
@@ -953,9 +1006,32 @@ def pattern_dict_items(pattern_dict, itempool=None, exhausted=None):
     for (key, value) in pattern_dict.items():
         if hasattr(value, 'item') and isinstance(value.item, list):
             if itempool is not None:
-                valid_items = [item.name for item in itempool if item.name in value.item]
+                valid_items = []
+                pool_group_items = []
+                for item in itempool:
+                    if item.name in value.item:
+                        valid_items.append(item.name)
+                    else:
+                        for group in item_groups:
+                            if '#' + group in value.item:
+                                if item.name in item_groups[group]:
+                                    pool_group_items.append(item.name)
                 if exhausted is not None:
-                    [valid_items.remove(item) for item in exhausted if item in valid_items]
+                    for used_item in exhausted:
```

Comment:
Would 'exhausted' be better named 'used_items'? I keep thinking it's supposed to refer to a list of items that can't be placed any more, with no repeats, but it seems to keep track of every individual item (potentially with repeats).

Reply:
That would make sense. Exhausted did actually work the former way previously as it was what was used to make sure every element in a plando list was used before it started repeating, but with some of the item groups being quite large and having repeatable values, it's seemed to become more just a list of what from the item pool has already been placed so that there isn't a bunch of overlap causing repeated items.